### PR TITLE
Input stream

### PR DESCRIPTION
**Pull request by ghost**
Allow lexicons to be populated via an input stream object.

### CODE REVIEW

**Comment by ghost** [comment]
I've tried to clean up the whitespace changes, but ran out of time.

File: src/simplenlg/lexicon/XMLLexicon.java [hunk 1]
```diff
@@ -230,51 +230,71 @@ public XMLLexicon(String languageCode, URI lexiconURI) {
...
-		indexByVariant = new HashMap<String, List<WordElement>>();
-		// added by vaudrypl
-		indexByCategory = new EnumMap<LexicalCategory, List<WordElement>>(LexicalCategory.class);
+     * Load an XML Lexicon from an Input Stream
+     * @param language
+     *            the associated language ISO 639-1 two letter code
+     * @param lexiconStream
+     * @author amanning
+     */
+    public XMLLexicon(Language language, InputStream lexiconStream) {
```
Comment (ghost) [inline src/simplenlg/lexicon/XMLLexicon.java:239]: Now the lexicon can be populated with an input stream which is more Android-friendly.

File: src/simplenlg/lexicon/XMLLexicon.java [hunk 2]
```diff
@@ -230,51 +230,71 @@ public XMLLexicon(String languageCode, URI lexiconURI) {
...
-	}
+	/**
+     * method to actually load and index the lexicon from a URI
+     *
+     * vaudrypl removed call to addSpecialCases() and moved this
+     * method to simplenlg.lexicon.english.XMLLexicon
+     *
+     * @param uri
+     */
+    private void createLexicon(Object lexicon) {
```
Comment (ghost) [inline src/simplenlg/lexicon/XMLLexicon.java:252]: This lack of typesafety isn't the best, but it is Java so there you go.

File: src/simplenlg/lexicon/XMLLexicon.java [hunk 3]
```diff
@@ -78,14 +79,14 @@
...
 
-	
+
 	/**********************************************************************/
 	// constructors
 	/**********************************************************************/
 
 	/**
 	 * Load an XML Lexicon from a named file
-	 *
```
Comment (ghost) [inline src/simplenlg/lexicon/XMLLexicon.java:88]: After working at trying to not pollute the history with these whitespace changes, I gave up and so they are in there.

**Comment by vaudrypl** [comment]
Sorry for the delay, but I must tell you that I will only have a few hours per week at best to spare to review your changes, so it might take some time to complete. On the other hand it means that you can take your time in cleaning and testing your code.

In particular, please note that this project follows the test-driven approach to development. That means that for every functionality that is added to the code in the _src_ folder, a corresponding test must be added to the _testsrc_ folder. (In fact, it is even better if the test is added before the implementation, because it provides a clear objective for the implementation.) In your case, you could add a test to _test.french.XMLLexiconTest.java_.

**Comment by ghost** [comment]
I'm going to work on the tests then re-submit.  Thanks for the quick review.  Hopefully I can get this out relatively soon.

**Pull request closed without merging**